palvelutarjotin/oidc.py

```diff
@@ -19,7 +22,31 @@ def validate_claims(self, id_token):
         super().validate_claims(id_token)
 
 
-class GraphQLApiTokenAuthentication(FixTunnistamoMixin, ApiTokenAuthentication):
+class FixLastLoginMixin:
+    def update_last_login_after_interval(self, user):
+        """
+        Update the user object last_login field if enough time has passed.
+        The setting UPDATE_LAST_LOGIN_INTERVAL determines whether or not
+        enough time has passed for update.
+        We don't want to update on every request,
+        because then there would be too much writes when there are lot of users.
+        """
+        minutes_since_last_update = None
+        if user.last_login:
+            minutes_since_last_update = (
+                timezone.now() - user.last_login
+            ).total_seconds() / 60
+        if (
+            minutes_since_last_update is None
+            or minutes_since_last_update
+            > settings.UPDATE_LAST_LOGIN["UPDATE_INTERVAL_MINUTES"]
```

Comment:
This way black formats Python code is horrible—meaning the indentation levels being all identical also for the or clause's innards—to me but it's probably still PEP8 compliant. I mean you can leave it like this because that's how black does it and we're using black for formatting but I would add parentheses around the or i.e. `or (minutes_since_last_update > settings.UPDATE_LAST_LOGIN["UPDATE_INTERVAL_MINUTES"])` to make black format the code more readably.

Reply:
black is black. Let's say that my interest to this PR is currently in zero, so I just take what the formatter gives.